src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableActivityContext.cs

```diff
@@ -129,7 +132,7 @@ internal object GetInput(Type destinationType)
             // MessagePayloadDataConverter to throw an exception. This is a workaround for that case. All other
             // inputs with destination System.String (in-proc: JSON and not JSON; out-of-proc: not-JSON) inputs with
             // destination System.String should cast to JValues and be handled above.)
-            if (destinationType.Equals(typeof(string)))
+            if (this.rawInput)
```

Comment:
retroactive bug find: this change in the if-condition affects the "legacy" OOProc SDKs - JS, PowerShell, Python. In those cases, the destinationType is `string`, so we should have entered this if-statement, but we no longer do.

Reply:
Ironically, the comment here describes precisely how this if statement was supposed to be used, but we missed that during review.